examples/vision/aodnet.py

```diff
@@ -0,0 +1,354 @@
+    def __init__(self, name: str, stddev: float = 0.02, weight_decay: float = 1e-4):
+        super(AODNet, self).__init__(name=name)
+        self.conv_layer_1 = tf.keras.layers.Conv2D(
+            filters=3,
+            kernel_size=1,
+            padding="same",
+            activation=tf.nn.relu,
+            kernel_initializer=tf.keras.initializers.RandomNormal(stddev=stddev),
+            kernel_regularizer=tf.keras.regularizers.L2(weight_decay),
+        )
+        self.conv_layer_2 = tf.keras.layers.Conv2D(
+            filters=3,
+            kernel_size=1,
+            padding="same",
+            activation=tf.nn.relu,
+            kernel_initializer=tf.keras.initializers.RandomNormal(stddev=stddev),
+            kernel_regularizer=tf.keras.regularizers.L2(weight_decay),
+        )
+        self.conv_layer_3 = tf.keras.layers.Conv2D(
+            filters=3,
+            kernel_size=5,
+            padding="same",
+            activation=tf.nn.relu,
+            kernel_initializer=tf.keras.initializers.RandomNormal(stddev=stddev),
+            kernel_regularizer=tf.keras.regularizers.L2(weight_decay),
+        )
+        self.conv_layer_4 = tf.keras.layers.Conv2D(
+            filters=3,
+            kernel_size=7,
+            padding="same",
+            activation=tf.nn.relu,
+            kernel_initializer=tf.keras.initializers.RandomNormal(stddev=stddev),
+            kernel_regularizer=tf.keras.regularizers.L2(weight_decay),
+        )
+        self.conv_layer_5 = tf.keras.layers.Conv2D(
+            filters=3,
+            kernel_size=3,
+            padding="same",
+            activation=tf.nn.relu,
+            kernel_initializer=tf.keras.initializers.RandomNormal(stddev=stddev),
+            kernel_regularizer=tf.keras.regularizers.L2(weight_decay),
+        )
+        self.relu = tf.keras.layers.ReLU(max_value=1.0)
+
+    def call(self, inputs, *args, **kwargs):
+        conv_1 = self.conv_layer_1(inputs)
+        conv_2 = self.conv_layer_2(conv_1)
+        concat_1 = tf.concat([conv_1, conv_2], axis=-1)
+        conv_3 = self.conv_layer_3(concat_1)
+        concat_2 = tf.concat([conv_2, conv_3], axis=-1)
+        conv_4 = self.conv_layer_4(concat_2)
+        concat_3 = tf.concat([conv_1, conv_2, conv_3, conv_4], axis=-1)
+        k = self.conv_layer_5(concat_3)
+        j = k * inputs - k + 1.0
+        output = self.relu(j)
+        return output
+
+
+"""
+Since we are training the model using a single GPU, i.e, Nvidia Tesla P100, we would use
+`tf.distribute.OneDeviceStrategy`, using this strategy will place any variables created
+in its scope on the specified device. We would define and compile our model in the scope
+of our distribution strategy. We would be using Adam as out optimizer, Mean Squared Error
+as the loss function and PSNR (Peak Signal Noise Ratio) as our metric.
+
+**Note:** For training AODNet in a multi-GPU environment,
+`tf.distribute.MirroredStrategy` can be used as a distribution stratgey.
```

Comment:
Suggestion:

- Add emphasis to "TensorFlow distribution strategy" (with this guide x 2: https://www.tensorflow.org/guide/distributed_training)
- Use the definite article "the"
- Fix a typo ("strategy")

```suggestion
In this example, we are training the model using a single GPU (an NVIDIA Tesla
P100) and the`tf.distribute.OneDeviceStrategy`. This
[TensorFlow distribution
strategy](https://www.tensorflow.org/tutorials/distribute/keras)
will place any variables created in its scope on the specified device. We will
define and compile our model in the scope of our distribution strategy, using
Adam as the optimizer, the Mean Squared Error
as the loss function, and the PSNR (Peak Signal Noise Ratio) as our metric.

**Note:** For training the AOD-Net in a multi-GPU environment, we can also use
`tf.distribute.MirroredStrategy` (go to the
[Distributed training](https://www.tensorflow.org/guide/distributed_training)
guide more information).
```